auth/routes.py

```diff
@@ -0,0 +1,95 @@
+from fastapi import APIRouter, Depends, HTTPException, status
+from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
+from sqlalchemy.orm import Session
+from auth.schemas import CreateUsers, Token, TokenData
+from auth.utils import verify_password, get_password_hash
+from auth.models import Users
+from database.database import get_db 
+from jose import jwt, JWTError
+from datetime import timedelta 
+
+# Import these from wherever they are defined
+from auth.utils import SECRET_KEY, ALGORITHM, ACCESS_TOKEN_EXPIRE_MINUTES
+
+oauth_2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+router = APIRouter()
+
+    
+def authenticate_user(db, username: str, password: str):
+
+    user = db.query(Users).filter(Users.username == username).first() 
+
+    if not user:
+        return False
+    
+    if not verify_password(password, user.hashed_password):
+        return False
+    
+    return user
+
+
+@router.post("/token", response_model=Token)
+def login_for_access_token(form_data: OAuth2PasswordRequestForm = Depends(), db: Session = Depends(get_db)):
+    user = authenticate_user(db, form_data.username, form_data.password)
+    if not user:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="Incorrect username or password",
+            headers={"WWW-Authenticate": "Bearer"},
+        )
+    access_token_expires = timedelta(minutes=ACCESS_TOKEN_EXPIRE_MINUTES)
+    access_token = jwt.encode(
+        {"sub": user.username}, SECRET_KEY, algorithm=ALGORITHM
+    )
+    return {"access_token": access_token, "token_type": "bearer"}
+
+
+
+@router.post("/registers")
```

Comment:
Why is this endpoint called registers? Isn't POST users a better name? As we're creating user here.